sdk/src/main/scala/com/horizen/forge/ForgeMessageBuilder.scala

```diff
@@ -313,7 +313,9 @@ class ForgeMessageBuilder(mainchainSynchronizer: MainchainSynchronizer,
         Seq()
       } else { // SC block is in the middle of the epoch
         var txsCounter: Int = 0
-        nodeView.pool.take(nodeView.pool.size).filter(tx => {
+        val allowedWithdrawalRequestBoxes = nodeView.state.getAllowedWithdrawalRequestBoxes(mainchainBlockReferenceDataToRetrieve.size) - nodeView.state.getAlreadyMinedWithdrawalRequestBoxesInCurrentEpoch
+        val mempoolTx = if (nodeView.state.backwardTransferLimitEnabled)  nodeView.pool.takeWithWithdrawalBoxesLimit(allowedWithdrawalRequestBoxes) else nodeView.pool.take(nodeView.pool.size)
```

Comment:
Be careful with the forger logic.
Let's assume that allowed WBs per MC block is 100, the state is before the Fork1, so no limit on BTs.
Mempool contains Tx with 200 WBs.
Then Forger creates the next block that reach the Fork1. So Forger will add Tx with 200 WBs, but state will reject this block, because 200 > 100 (BT limit).
Proposal:
* remove `nodeView.state.backwardTransferLimitEnabled` method, and use fork manager directly in the state and forger.
* for forger get the consensus epoch number of the candidate block -> pass it to the ForkManager to make a decision if `backwardTransferLimitEnabled` is enabled.
Let's add a separate python test to verify this tricky case.